mitmproxy/net/http/request.py

```diff
@@ -367,13 +367,64 @@ def _get_urlencoded_form(self):
                 pass
         return ()
 
+    def _sanitized_form_data_replacement(self, form_data):
+        # check if unmodified content changes with URL encoding
+        url_decoded_content = urllib.parse.parse_qsl(self.content, keep_blank_values=True)
+        url_reencoded_content = urllib.parse.urlencode(url_decoded_content, doseq=True).encode()
+        if self.content == url_reencoded_content:
```

Comment:
Why do we need to special-case this? None of our heuristics should trigger if it's "normally" urlencoded.